cmd/cluster/network/endpoint/update_endpoint.go

```diff
@@ -0,0 +1,86 @@
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package endpoint
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	ybmAuthClient "github.com/yugabyte/ybm-cli/internal/client"
+	ybmclient "github.com/yugabyte/yugabytedb-managed-go-client-internal"
+)
+
+var updateEndpointCmd = &cobra.Command{
+	Use:   "update",
+	Short: "Update a network endpoint for a cluster",
+	Long:  `Update a network endpoint for a cluster`,
+	Run: func(cmd *cobra.Command, args []string) {
+		authApi, err := ybmAuthClient.NewAuthApiClient()
+		if err != nil {
+			logrus.Fatalf("Could not initiate api client: %s", err.Error())
+		}
+		authApi.GetInfo("", "")
+
+		clusterEndpoints, clusterId, endpointId := getEndpointById(cmd, authApi)
+
+		// We currently support fetching just Private Service Endpoints
+		switch clusterEndpoints[0].GetAccessibilityType() {
+
+		case ybmclient.ACCESSIBILITYTYPE_PRIVATE_SERVICE_ENDPOINT:
+			if !cmd.Flags().Changed("security-principals") {
+				logrus.Fatalf("security-principals is required for private service endpoints")
+			}
+
+			pseGetResponse, r, err := authApi.GetPrivateServiceEndpoint(clusterId, endpointId).Execute()
+			if err != nil {
+				logrus.Debugf("Full HTTP response: %v", r)
+				logrus.Fatalf("Error when calling `ClusterApi.GetPrivateServiceEndpoint`: %s", ybmAuthClient.GetApiErrorDetails(err))
+			}
+
+			securityPrincipalsString, _ := cmd.Flags().GetString("security-principals")
+			securityPrincipalsList := strings.Split(securityPrincipalsString, ",")
+
+			regionArnMap := make(map[string][]string)
+			regionArnMap[pseGetResponse.Data.Spec.ClusterRegionInfoId] = securityPrincipalsList
+
+			// we create a spec that has a single element
+			pseSpec := authApi.CreatePrivateServiceEndpointRegionSpec(regionArnMap)
+
+			// we pass the only element in the spec to the update endpoint call
+			updateResp, r, err := authApi.EditPrivateServiceEndpoint(clusterId, endpointId).PrivateServiceEndpointRegionSpec(pseSpec[0]).Execute()
+			if err != nil {
+				logrus.Debugf("Full HTTP response: %v", r)
+				logrus.Fatalf("Error when calling `ClusterApi.EditPrivateServiceEndpoint`: %s", ybmAuthClient.GetApiErrorDetails(err))
+			}
+
+			msg := fmt.Sprintf("Updated endpoint %s", updateResp.Data.Info.Id)
+			fmt.Println(msg)
+
+		default:
+			logrus.Fatalf("Endpoint is not a private service endpoint. Only private service endpoints are currently supported.")
+		}
+
+	},
+}
+
+func init() {
+	EndpointCmd.AddCommand(updateEndpointCmd)
+	updateEndpointCmd.Flags().String("endpoint-id", "", "[REQUIRED] The ID of the endpoint")
```

Comment:
again, more of a question: if there is a single endpoint per region, can we use the region?

Reply:
See above.